python/hsfs/core/tfdata_engine.py

```diff
@@ -0,0 +1,484 @@
+#
+#   Copyright 2020 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import re
+import itertools
+
+try:
+    import tensorflow as tf
+except ModuleNotFoundError:
+    pass
+
+try:
+    from pydoop import hdfs
+except ModuleNotFoundError:
+    pass
+
+try:
+    import boto3
+except ModuleNotFoundError:
+    pass
+
+
+class TFDataEngine:
+    SUPPORTED_TFDTYPES = [
+        tf.float16,
+        tf.float32,
+        tf.float64,
+        tf.int16,
+        tf.int64,
+        tf.int32,
+    ]
+    SPARK_TO_TFDTYPES_MAPPINGS = {
+        "string": tf.string,
+        "short": tf.int16,
+        "int": tf.int32,
+        "long": tf.int64,
+        "float": tf.float32,
+        "double": tf.float64,
+    }
+
+    def __init__(
+        self,
+        training_dataset,
+        split,
+        target_name,
+        feature_names,
+        is_training,
+        cycle_length,
+    ):
+        """TFDataEngine object that has utility methods for tf.data
+
```

Comment:
you can add a longer description with a quick example here, eg:

!!! example "TFRecordDataset"

    ```python
    train_input_feeder = training_dataset.tf_data(target_name="label",
                                                  split="train",
                                                  is_training=True)
    train_input = train_input_feeder.tf_record_dataset()
    ```

the indentation is important for it to work, github is previewing it weirdly but after the `!!!` line, everything that's indented will be shown within the example tab